src/FunderMaps.Data/Repositories/AnalysisRepository.cs

```diff
@@ -147,57 +120,41 @@ public async Task<AnalysisProduct> GetByIdInFenceAsync(Guid userId, string id, C
         {
             id.ThrowIfNullOrEmpty();
 
-            // FUTURE This always returns EntityNotFoundException when the building is either not-existent.
             var sql = @"
-                SELECT
-                    ac.id,
-                    ac.external_id,
-                    ac.external_source,
-                    ac.foundation_type,
-                    ac.groundwater_level,
-                    ac.foundation_risk,
-                    ac.construction_year,
-                    ac.building_height,
-                    ac.ground_level,
-                    ac.restoration_costs,
-                    ac.dewatering_depth,
-                    ac.drystand,
-                    ac.reliability,
-                    ac.neighborhood_id
-                FROM data.analysis_product_complete AS ac
-                WHERE
-                    ac.id = @id";
-
-            await using var connection = await DbProvider.OpenConnectionScopeAsync(token);
-            await using var cmd = DbProvider.CreateCommand(sql, connection);
-            cmd.AddParameterWithValue("id", id);
-
-            await using var reader = await cmd.ExecuteReaderAsyncEnsureRowAsync();
-            await reader.ReadAsync(token);
+                SELECT  -- AnalysisProduct
+                        ac.id,
+                        ac.external_id,
+                        ac.external_source,
+                        ac.foundation_type,
+                        ac.groundwater_level,
+                        ac.foundation_risk,
+                        ac.construction_year,
+                        ac.building_height,
+                        ac.ground_level,
+                        ac.restoration_costs,
+                        ac.dewatering_depth,
+                        ac.drystand,
+                        ac.reliability,
+                        ac.neighborhood_id
+                FROM    data.analysis_product_complete AS ac
+                WHERE   id = @id
+                LIMIT   1";
+
+            await using var context = await DbContextFactory(sql);
+
+            context.AddParameterWithValue("id", id);
+
+            await using var reader = await context.ReaderAsync();
 
             return MapFromReader(reader);
         }
 
-        // FUTURE Sorting order and sorting column
-        /// <summary>
-        ///     
-        /// </summary>
-        /// <param name="userId">Internal user id.</param>
-        /// <param name="query"></param>
-        /// <param name="navigation"></param>
-        /// <param name="token"></param>
-        /// <returns></returns>
-        public async Task<IEnumerable<AnalysisProduct>> GetByQueryAsync(
-            Guid userId,
-            string query,
-            INavigation navigation,
-            CancellationToken token = default)
+        public async IAsyncEnumerable<AnalysisProduct> GetBySearchQueryAsync(Guid userId, string query, INavigation navigation)
```

Comment:
Maybe make a `GetBySearchQueryInFenceAsync` and a `GetBySearchQueryAsync` call here as well.

Reply:
Conclusion: The context should be used to determine whether or not a fence should be added.